source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/TextBlockRenderer.java

```diff
@@ -225,32 +225,26 @@ private void applyAccessibilityProperties(TextView textView, Context context, Re
         }
         textView.setImportantForAccessibility(View.IMPORTANT_FOR_ACCESSIBILITY_YES);
 
-        if (textHandleResult.getHasLinks())
-        {
+        if (textHandleResult.getHasLinks()) {
             textView.setMovementMethod(LinkMovementMethod.getInstance());
-            if (textHandleResult.isALink())
-            {
+            if (textHandleResult.isALink()) {
                 textView.setOnKeyListener(new SingleLinkOnKeyListener(new SpannableString(textHandleResult.getHtmlString())));
-            }
-        }
 
-        AccessibilityManager am = (AccessibilityManager)context.getSystemService(ACCESSIBILITY_SERVICE);
-        am.addAccessibilityStateChangeListener(new AccessibilityManager.AccessibilityStateChangeListener() {
-            @Override
-            public void onAccessibilityStateChanged(boolean b)
-            {
-                boolean isEnabled = am.isEnabled();
-                if (b && isEnabled)
-                {
-                    textView.setFocusable(true);
-                }
-                else
-                {
-                    textView.setFocusable(false);
-                }
+                AccessibilityManager am = (AccessibilityManager) context.getSystemService(ACCESSIBILITY_SERVICE);
+                am.addAccessibilityStateChangeListener(new AccessibilityManager.AccessibilityStateChangeListener() {
+                    @Override
+                    public void onAccessibilityStateChanged(boolean b) {
+                        boolean isEnabled = am.isEnabled();
+                        if (b && isEnabled) {
+                            textView.setFocusable(true);
+                        } else {
+                            textView.setFocusable(false);
+                        }
+                    }
+                });
+                textView.setFocusable(am.isEnabled());
```

Comment:
can you verify that, by moving this line here, cases such as 

"this is a string [link](url) and this is another [link](url)"

also behaves as expected?

Reply:
Yes, I'll check on this

Reply:
Hey @almedina-ms, I tested this scenario, and it does not work as expected.

I can move the focus logic to be within the `hasLinks` block; however, the individual spans are not accessible via tab key and pressing enter does not launch either url. 

I am seeing this issue on main as well, so I will file a separate bug for this. Seems like the problem here is that `URLSpan` is not a view, so we cannot use the `SetFocus` method.

Reply:
Created issue #8077